src/main/java/ar/com/synergian/wagongit/Utils.java

```diff
@@ -11,15 +11,14 @@
 	private Utils() {
 	}
 
-	public static File createCheckoutDirectory(String path, boolean isPermanent) throws GitException {
+	public static File createCheckoutDirectory(String path) throws GitException {
 		File dir;
-		if (isPermanent) {
-			dir = new File(path);
-		} else {
+		if (path.endsWith(".git")) {
 			dir = new File(System.getProperty("java.io.tmpdir"), "wagon-git-" + hashPath(path));
+		} else {
+			dir = new File(path);
```

Comment:
So, about this flag, should we keep it and enable it only when wagon.git.permanent.root is not set ? (ie. when wagon.git.permanent.root is set, wagon.git.safe.checkout is  considered as false).